quakesaver_client/models/local_sensor.py

```diff
@@ -0,0 +1,180 @@
+        """
+        ...
+
+    def _get_measurement(
+        self: LocalSensor, query: MeasurementQueryFull
+    ) -> MeasurementResult:
+        """Request measurements of the sensor."""
+        ...
+
+    def get_peak_horizontal_acceleration(
+        self: LocalSensor, query: MeasurementQuery
+    ) -> MeasurementResult:
+        """Get the PGA measurement of the sensor.
+
+        Args:
+            query: The query parameters like time frame and aggregator.
+
+        Returns:
+            MeasurementQuery: The queried data (if exists) as time series.
+        """
+        ...
+
+    def get_jma_intensity(
+        self: LocalSensor, query: MeasurementQuery
+    ) -> MeasurementResult:
+        """Get the JMA Intensity measurement of the sensor.
+
+        Args:
+            query: The query parameters like time frame and aggregator.
+
+        Returns:
+            MeasurementQuery: The queried data (if exists) as time series.
+        """
+        ...
+
+    def get_rms_amplitude(
+        self: LocalSensor, query: MeasurementQuery
+    ) -> MeasurementResult:
+        """Get the RMS Amplitude measurement of the sensor.
+
+        Args:
+            query: The query parameters like time frame and aggregator.
+
+        Returns:
+            MeasurementQuery: The queried data (if exists) as time series.
+        """
+        ...
+
+    def get_spectral_intensity(
+        self: LocalSensor, query: MeasurementQuery
+    ) -> MeasurementResult:
+        """Get the Spectral Intensity measurement of the sensor.
+
+        Args:
+            query: The query parameters like time frame and aggregator.
+
+        Returns:
+            MeasurementQuery: The queried data (if exists) as time series.
+        """
+        ...
+
+    def get_rms_offset(self: LocalSensor, query: MeasurementQuery) -> MeasurementResult:
+        """Get the RMS Offset measurement of the sensor.
+
+        Args:
+            query: The query parameters like time frame and aggregator.
+
+        Returns:
+            MeasurementQuery: The queried data (if exists) as time series.
+        """
+        ...
+
+    def get_waveform_stream(self: LocalSensor) -> WebsocketHandler:
+        """Get a `WebsocketHandler` to server waveform data."""
+        return WebsocketHandler(self.url)
+
+    def get_waveform_data(
+        self: LocalSensor,
+        start_time: datetime,
+        end_time: datetime,
+        location_to_store: Path | str = None,
+    ) -> Path | None:
+        """Request FDSN waveform dat of the sensor."""
+        ...
+
+    def get_stationxml(
+        self: LocalSensor,
+        start_time: datetime,
+        end_time: datetime,
+        minlatitude: float = -90,
+        maxlatitude: float = 90,
+        minlongitude: float = -180,
+        maxlongitude: float = 180,
+        level: StationDetailLevel = "station",
+        location_to_store: Path | str = None,
+    ) -> Path:
+        """Request FDSN StationXML metadata of the sensor."""
+        ...
+
```

Comment:
Add `async def get_waveforms(start: datetime, end: datetime, channel: str = '*')`?

Reply:
That will be a separate MR.